src/browser/setupWorker/start/createRequestListener.ts

```diff
@@ -30,6 +30,8 @@ export const createRequestListener = (
     const request = parseWorkerRequest(message.payload)
     const requestCloneForLogs = request.clone()
 
+    context.requests.set(requestId, request.clone())
```

Comment:
I'm not sure what I feel in particular about this approach but I've been meaning to provide in-memory requestId/request storage for some time now. This will simplify the usage of life-cycle events.